src/presenter/films-presenter.js

```diff
@@ -1,102 +1,150 @@
+  #sortComponent = null;
 
-  #sectionMovie = [];
   #movieModel = null;
   #filmListContainer = null;
   #renderedMovieCount = SHOW_FILM_COUNT_STEP;
 
   #filmPresenter = new Map();
   #currentSortType = SortType.DEFAULT;
-  #sourcedFilms = [];
+  #filterType = FilterType.ALL;
 
-  constructor(filmListContainer, movieModel) {
+  #filterModel = null;
+
+  constructor(filmListContainer, movieModel, filterModel) {
     this.#filmListContainer = filmListContainer;
     this.#movieModel = movieModel;
+    this.#filterModel = filterModel;
+
+    this.#movieModel.addObserver(this.#handleModelEvent);
+    this.#filterModel.addObserver(this.#handleModelEvent);
+  }
+
+  get movies() {
+    this.#filterType = this.#filterModel.filter;
+    const movies = this.#movieModel.movies;
+    const filteredMovies = filter[this.#filterType](movies);
+
+    switch (this.#currentSortType) {
+      case SortType.RATING:
+        return filteredMovies.sort(sortFilmsByRating);
+      case SortType.DATE:
+        return filteredMovies.sort(sortFilmsByDate);
+    }
+
+    return filteredMovies;
   }
 
   init = () => {
-    this.#sectionMovie = [...this.#movieModel.movie];
-    this.#sourcedFilms = [...this.#movieModel.movie];
     this.#renderMovie();
   };
 
   #handleLoadMoreButtonClick = () => {
-    this.#renderFilms(this.#renderedMovieCount, this.#renderedMovieCount + SHOW_FILM_COUNT_STEP);
+    const movieCount = this.movies.length;
+    const newRenderedMovieCount = Math.min(movieCount, this.#renderedMovieCount + SHOW_FILM_COUNT_STEP);
+    const movies = this.movies.slice(this.#renderedMovieCount, newRenderedMovieCount);
 
-    this.#renderedMovieCount += SHOW_FILM_COUNT_STEP;
+    this.#renderFilms(movies);
+    this.#renderedMovieCount = newRenderedMovieCount;
 
-    if (this.#renderedMovieCount >= this.#sectionMovie.length) {
+    if (this.#renderedMovieCount >= movieCount) {
       remove(this.#loadMoreButtonComponent);
     }
   };
 
-  #renderFilms = (from, to) => {
-    this.#sectionMovie
-      .slice(from, to)
-      .forEach((element) => this.#createFilm(element));
+  #renderFilms = (movies) => {
+    movies.forEach((element) => this.#createFilm(element));
   };
 
   #renderNoFilms = () => {
-    render(this.#noFilmComponent, this.#filmListContainer);
+    this.#noFilmComponent = new NoFilmCardView(this.#filterType);
+    render(this.#noFilmComponent, this.#filmSection.element);
   };
 
   #renderLoadMoreButton = () => {
-    render(this.#loadMoreButtonComponent, this.#filmSection.element);
+    this.#loadMoreButtonComponent = new LoadMoreButtonView();
     this.#loadMoreButtonComponent.setClickLoadHandler(this.#handleLoadMoreButtonClick);
+
+    render(this.#loadMoreButtonComponent, this.#filmSection.element);
   };
 
-  #clearFilmList = () => {
+  #clearFilm = ({resetRenderedMovieCount = false, resetSortType = false} = {}) => {
+    const movieCount = this.movies.length;
+
     this.#filmPresenter.forEach((presenter) => presenter.destroy());
```

Comment:
Немного поменяем логику очистки: теперь будем проверять, открыто ли модальное окно
```suggestion
        this.#filmPresenter.forEach((presenter) => {
      if (presenter.isOpened) {
        presenter.destroyOnlyCard();
        this.#openedFilmPresenter = presenter;
      } else {
        presenter.destroy();
      }
    });
```